src/converter/internal/convertercontroller.cpp

```diff
@@ -101,15 +102,35 @@ mu::Ret ConverterController::fileConvert(const io::path_t& in, const io::path_t&
 
     globalContext()->setCurrentProject(notationProject);
 
+    if (suffix == engraving::MSCZ || suffix == engraving::MSCX) {
+        return notationProject->save(out);
+    }
+
+    io::path_t targetDirPath = io::absoluteDirpath(out);
+    if (!io::File::exists(targetDirPath)) {
+        io::Dir targetDir(targetDirPath);
+        ret = targetDir.mkpath(targetDir.absolutePath());
+        if (!ret) {
+            LOGE() << "Failed to make directory, err: " << ret.toString() << ", path: " << targetDirPath;
+            return ret;
+        }
+    }
```

Comment:
I wonder a little bit about whether this is really necessary. I believe that usually when you call `open` on a non-existent file, it creates the entire path to that file, i.e. it also creates any missing folders.
If there are specific export formats that use a different way of writing that does fail in the case of non-existent folders, we might want to put this code in the writers for those formats themselves, instead of here.